### PR TITLE
Move and rename assignment to students_metrics

### PR DESCRIPTION
**Pull request by marcospri**
This better reflects what the endpoint returns.

See: https://hypothes-is.slack.com/archives/C4K6M7P5E/p1719996931602029

## Testing

Navigate all three levels starting from courses: 

http://localhost:8001/dashboard/organizations/local.lms.org.LDtcl7EUTeW2UERPJLAVtA

### CODE REVIEW

File: lms/js_config_types.py [hunk 1]
```diff
@@ -76,7 +76,7 @@ class DashboardRoutes(TypedDict):
     assignment: str
     """Fetch a single assigment by ID"""
 
-    assignment_stats: str
```
Comment (marcospri) [inline lms/js_config_types.py:79]: Renamed

File: lms/routes.py [hunk 2]
```diff
@@ -268,14 +268,15 @@ def includeme(config):  # noqa: PLR0915
     config.add_route(
         "api.dashboard.assignment", "/api/dashboard/assignments/{assignment_id}"
     )
-    config.add_route(
```
Comment (marcospri) [inline lms/routes.py:271]: Renamed and moved close to the other /students endpoint.

File: lms/static/scripts/frontend_apps/components/dashboard/test/AssignmentActivity-test.js [hunk 3]
```diff
@@ -56,7 +56,7 @@ describe('AssignmentActivity', () => {
       dashboard: {
         routes: {
           assignment: '/api/assignments/:assignment_id',
-          assignment_stats: '/api/assignments/:assignment_id/stats',
+          students_metrics: '/api/students/:assignment_id/metrics',
```
Comment (marcospri) [inline lms/static/scripts/frontend_apps/components/dashboard/test/AssignmentActivity-test.js:59]: Keeping the :assignment_id: in the URL in this PR.

File: lms/static/scripts/frontend_apps/config.ts [hunk 4]
```diff
@@ -251,8 +251,8 @@ export type EmailPreferences = {
 export type DashboardRoutes = {
   /** Fetch a single assignment by ID */
   assignment: string;
-  /** Fetch students stats for a single assignment by ID */
-  assignment_stats: string;
+  /** Fetch students metrics for a single assignment by ID */
```
Comment (marcospri) [inline lms/static/scripts/frontend_apps/config.ts:254]: This is still single assignment.

File: lms/views/dashboard/api/assignment.py [hunk 5]
```diff
@@ -65,55 +62,3 @@ def assignment(self) -> APIAssignment:
             title=assignment.title,
             course=APICourse(id=assignment.course.id, title=assignment.course.lms_name),
         )
-
-    @view_config(
```
Comment (marcospri) [inline lms/views/dashboard/api/assignment.py:69]: Moved to /api/user

File: lms/views/dashboard/api/user.py [hunk 6]
```diff
@@ -57,3 +61,55 @@ def students(self) -> APIStudents:
             ],
             "pagination": pagination,
         }
+
+    @view_config(
+        route_name="api.dashboard.students.metrics",
```
Comment (marcospri) [inline lms/views/dashboard/api/user.py:66]: Copied from the other module, just changed the route_name.

**Review by acelaya — APPROVED** [review]

**Pull request merged**